corehq/ex-submodules/casexml/apps/phone/restore.py

```diff
@@ -605,8 +605,11 @@ def get_response(self):
             response = HttpResponse(response, content_type="text/xml; charset=utf-8",
                                     status=412)  # precondition failed
         except CannotRestoreException as e:
-            response = get_simple_response_xml(str(e), ResponseNature.OTA_RESTORE_ERROR)
-            response = HttpResponse(response, content_type="text/xml; charset=utf-8", status=400)
+            # mobile expects a json response for 406 errors
+            response = JsonResponse({
+                'error': 'restore.failed.error',
+                'default_response': str(e),
+            })
```

Comment:
Not sure what the default status code of `JsonResponse` is, but guessing it's not 406. Does it accept a status code parameter? Something like
```suggestion
            }, status=406)
```